jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/EnvironmentBuilder.java

```diff
@@ -61,6 +60,6 @@ public void addClassPath(String... classPaths)
 
     public Environment build() throws Exception
     {
-        return new Environment.Named(_name, new URLClassLoader(_classpath.toArray(new URL[0]), EnvironmentBuilder.class.getClassLoader()));
+        return Environment.ensure(_name, new URLClassLoader(_classpath.toArray(new URL[0]), EnvironmentBuilder.class.getClassLoader()));
```

Comment:
What if the name already exists? This will just discard the classloader 
We need at least an error

Reply:
`ensure` does not allow that. This doesn't look like it's the right contract for the job here.

Let me think of something else...